accepted/color-4-new-spaces.md

```diff
@@ -1782,18 +1782,10 @@ is-missing($color, $channel)
 same($color1, $color2)
 ```
 
-* If either `$color1` or `$color2` is not a color in a [known color space]:
+* Let `color1` and `color2` be the result of [converting] `$color1` and
+  `$color2` into `xyz` color space, respectively.
 
-  * Let `color1` be `$color1`, and let `color2` be `$color2`.
-
-  > We can compare, but we can't do conversion. The color space remains
-  > relevant to equality. While this is technically the same as using `==`,
-  > it makes the function more robust to allow comparison of all colors.
-
-* Otherwise:
-
-  * Let `color1` and `color2` be the result of [converting] `$color1` and
-    `$color2` into `xyz` color space, respectively.
+* Replace any [missing] components in `color1` and `color2` with 0.
```

Comment:
What's the reasoning behind this? Just that they would result in the same display value?

Reply:
Here's my reasoning:

1. Missing channels are defined by spec as being treated as 0 everywhere other than interpolation. Since this is not an interpolation context, I think it fits the conceptual model to treat them as 0 here as well.

2. Missing channels are only well-defined _within a given color space_, and `same()` is trying to provide uniform behavior _across color spaces_. If we treat missing channels as distinct here, then there exists _no_ Oklch color that's `same()` as `rgb(none 150 255)`.

3. Relatedly, the previous definition already didn't consider missing channels in polar spaces (the place where they're by far the most meaningful and likely to arise naturally) as meaningful. Because it converts both colors into `xyz`, a missing hue is always treated identically to an undefined hue—which I think it good, because we want `color.same(oklch(50% 0% 0deg), oklch(50% 0% 180deg))` to return `true` since they're both visually gray.

Reply:
@nex3 Did you see the recent discussions around preserving `none` in calculations whenever possible? https://github.com/w3c/csswg-drafts/issues/10211#issuecomment-2075435167

Reply:
Interesting! I hadn't seen that. It does weaken point 1 somewhat, although I think in general the other points stand. I think making the distinction "`==` compares every little detail, `same()` compares whether the colors will look the same if you use them directly" makes sense.

Reply:
Then I'm OK with it, but I think a comment should be added to make it clear that this `same()` is meant for colors that are displayed the same